spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8ConfigMapPropertySourceLocator.java

```diff
@@ -45,7 +45,7 @@ public class Fabric8ConfigMapPropertySourceLocator extends ConfigMapPropertySour
 
 	Fabric8ConfigMapPropertySourceLocator(KubernetesClient client, ConfigMapConfigProperties properties,
 			KubernetesNamespaceProvider provider) {
-		super(properties);
+		super(properties, new Fabric8ConfigMapsCache());
```

Comment:
pass the cache implementation, so that in the super class we could call `discardAll`. This is what I was saying that if we don't want to clean the cache, we would not need to pass anything to super call. But that would mean hanging on to those entries forever, which is not ideal, imo.